src/quicklist.c

```diff
@@ -476,12 +513,32 @@ REDIS_STATIC int _quicklistNodeAllowMerge(const quicklistNode *a,
         (node)->sz = ziplistBlobLen((node)->zl);                               \
     } while (0)
 
+static void __quicklistInsertEntryNode(quicklist *quicklist, quicklistNode *old_node,
+                                       void *value, size_t sz, bool after)  {
+
+
+    quicklistNode *new_node = quicklistCreateNode();
+    new_node->entry = zmalloc(sz);
+    new_node->container = QUICKLIST_NODE_CONTAINER_NONE;
+    memcpy(new_node->entry , value, sz);
+    new_node->sz = sz;
```

Comment:
i saw you added a union alias, let's use it or drop it.
```suggestion
    new_node->entry = sz;
```

Reply:
not sure I understand the union is union {unsigned char *zl; unsigned char *entry;}; 
the size is common for both entry and zl. do you think I should add also a union for the size ?

Reply:
sorry. seems that i thought you're writing int `zl` in this line.
ignore..